modules/apps/portal-security/portal-security-script-management-web/src/main/java/com/liferay/portal/security/script/management/web/internal/groovy/script/uses/factory/GroovyScriptUsesFactoryRegistryImpl.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.security.script.management.web.internal.groovy.script.uses.factory;
+
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerList;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerListFactory;
+import com.liferay.portal.kernel.db.partition.DBPartition;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.security.script.management.groovy.script.use.GroovyScriptUse;
+import com.liferay.portal.security.script.management.groovy.script.uses.factory.GroovyScriptUsesFactory;
+import com.liferay.portal.security.script.management.groovy.script.uses.factory.GroovyScriptUsesFactoryRegistry;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(service = GroovyScriptUsesFactoryRegistry.class)
+public class GroovyScriptUsesFactoryRegistryImpl
+	implements GroovyScriptUsesFactoryRegistry {
+
+	@Override
+	public List<GroovyScriptUse> getGroovyScriptUses(Locale locale)
+		throws Exception {
+
+		List<GroovyScriptUse> groovyScriptUses = new ArrayList<>();
+
+		if (DBPartition.isPartitionEnabled()) {
+			_companyLocalService.forEachCompanyId(
```

Comment:
Hi @MarinhoFeliphe , does the `_companyLocalService.forEachCompanyId` approach also works when `DBPartition.isPartitionEnabled()` is false, because if it does then maybe we can remove the `if`. I believe that we don't have many companies when the db is not partitioned to have a performance gain in that case.
Let me know your thoughts.